File: BookWorm-Dotnet/Controllers/OtpController.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using BookWorm_Dotnet.Services;

namespace BookWorm_Dotnet.Controllers
{
    [ApiController]
    [Route("api/otp")]
    public class OtpController : ControllerBase
    {
        private readonly IOtpService _otpService;
        private readonly IEmailService _emailService;

        public OtpController(IOtpService otpService, IEmailService emailService)
        {
            _otpService = otpService;
            _emailService = emailService;
        }

        [HttpPost("send-otp")]
        public async Task<IActionResult> SendOtp([FromBody] Dictionary<string, string> request)
        {
            if (!request.ContainsKey("email"))
                return BadRequest(new { Message = "Email is required" });

            string email = request["email"];
            string otp = _otpService.GenerateOtp(email);
            await _emailService.SendOtpEmailAsync(email, otp);

            return Ok(new { Message = "OTP sent successfully" });
        }
        [HttpPost("verify-otp")]
        public IActionResult VerifyOtp([FromBody] Dictionary<string, string> request)
        {
            if (!request.ContainsKey("email") || !request.ContainsKey("otp"))
                return BadRequest(new { Message = "Email and OTP are required" });

            string email = request["email"];
            string otp = request["otp"];

            bool isValid = _otpService.ValidateOtp(email, otp);

            if (isValid)
                return Ok(new { Message = "OTP verified successfully" });
            else
                return BadRequest(new { Message = "Invalid or expired OTP" });
        }
    }
}

File: BookWorm/src/main/java/com/Project/BookWorm/Controller/OtpController.java
package com.Project.BookWorm.Controller;

import com.Project.BookWorm.Service.EmailService;
import com.Project.BookWorm.Service.OtpService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.HashMap;
import java.util.Map;

@RestController
@RequestMapping("/api")
@CrossOrigin("*")
public class OtpController {

    @Autowired
    private OtpService otpService;
    
    @Autowired
    private EmailService emailService;

    @PostMapping("/send-otp")
    public Map<String, String> sendOtp(@RequestBody Map<String, String> request) {
        String email = request.get("email");
        String otp = otpService.generateOtp(email);
        emailService.sendOtpEmail(email, otp);

        Map<String, String> response = new HashMap<>();
        response.put("message", "OTP sent successfully");
        return response;
    }

    @PostMapping("/verify-otp")
    public Map<String, String> verifyOtp(@RequestBody Map<String, String> request) {
        String email = request.get("email");
        String otp = request.get("otp");

        Map<String, String> response = new HashMap<>();
        if (otpService.validateOtp(email, otp)) {
            response.put("message", "OTP verified successfully");
        } else {
            response.put("message", "Invalid OTP");
        }
        return response;
    }
}



